WalletWasabi.Fluent/ViewModels/Wallets/Send/LabelSelectionViewModel.cs

```diff
@@ -32,18 +33,35 @@ public LabelSelectionViewModel(Money targetAmount)
 
 	public IEnumerable<LabelViewModel> LabelsBlackList => AllLabelsViewModel.Where(x => x.IsBlackListed);
 
-	public Pocket[] AutoSelectPockets()
+	public Pocket[] AutoSelectPockets(SmartLabel recipient)
 	{
 		var knownPockets = NonPrivatePockets.Where(x => x.Labels != CoinPocketHelper.UnlabelledFundsText).ToArray();
 		var unknownPockets = NonPrivatePockets.Except(knownPockets).ToArray();
 		var privateAndUnknownPockets = _allPockets.Except(knownPockets).ToArray();
 		var privateAndKnownPockets = _allPockets.Except(unknownPockets).ToArray();
+		var knownByRecipientPockets = knownPockets.Where(pocket => pocket.Labels.Any(label => recipient.Contains(label, StringComparer.OrdinalIgnoreCase))).ToArray();
+		var onlyKnownByRecipientPockets =
+			knownByRecipientPockets
+				.Where(pocket =>
+					pocket.Labels.Count() == recipient.Count() &&
+					pocket.Labels.All(label => recipient.Contains(label, StringComparer.OrdinalIgnoreCase)))
+				.ToArray();
+
+		if (onlyKnownByRecipientPockets.Sum(x => x.Amount) >= _targetAmount)
+		{
+			return onlyKnownByRecipientPockets;
+		}
 
 		if (_privatePocket.Amount >= _targetAmount)
 		{
 			return new[] { _privatePocket };
 		}
 
+		if (GetBestKnownByRecipientPockets(knownByRecipientPockets, _targetAmount, recipient) is { } pockets)
```

Comment:
I don't prefer this, because if someone used the `GetBestKnownByRecipientPockets` method without checking the amount before calling it, the method could return pockets that do not cover `_targetAmount`. For safety reasons, the amount check should be inside of the method.

I modified it. It does not do unnecessary calculations and I removed the nullable.